circuit_maintenance_parser/parsers/tata.py

```diff
@@ -0,0 +1,69 @@
+# pylint: disable=disallowed-name
+"""Circuit maintenance parser for Tata Email notifications."""
+from typing import List, Dict, Any
+from datetime import datetime
+
+from bs4.element import ResultSet  # type: ignore
+from circuit_maintenance_parser.output import Impact, Status
+from circuit_maintenance_parser.parser import Html, EmailSubjectParser
+
+
+class HtmlParserTata(Html):
+    """Custom Parser for HTML portion of Tata circuit maintenance notifications."""
+
+    def parse_html(self, soup: ResultSet) -> List[Dict]:
+        """Parse Tata circuit maintenance email."""
+        prev: str = ""
+        data: Dict[str, Any] = {
+            "account": "N/A",
+            "circuits": [],
+            "organizer": soup.select("a[href^=mailto]")[0].text.strip(),
+        }
+        for span in soup.find_all("span"):
+            curr = span.text.strip()
+            if curr != prev:
+                if prev.lower() == "ticket reference - tcl":
+                    data["maintenance_id"] = curr
+                elif prev.lower() == "service id":
+                    for circuit in curr.split(","):
+                        data["circuits"].append(
+                            {
+                                "circuit_id": circuit.strip(),
+                                "impact": Impact.OUTAGE,
+                            }
+                        )
+                elif prev.lower() == "activity window (gmt)" or prev.lower() == "revised activity window (gmt)":
+                    start_end = curr.split("to")
+                    data["start"] = self._parse_time(start_end[0])
+                    data["end"] = self._parse_time(start_end[1])
+                elif "extended up to time window" in prev.lower():
+                    if "gmt" in curr.lower():
+                        data["end"] = self._parse_time(curr)
+            prev = span.text.strip()
+
+        return [data]
+
+    @staticmethod
+    def _parse_time(string: str) -> int:
+        """Convert YYYY-MM-DD HH:MM:SS GMT to epoch."""
+        return int((datetime.strptime(string.strip(), "%Y-%m-%d %H:%M:%S GMT") - datetime(1970, 1, 1)).total_seconds())
+
+
+class SubjectParserTata(EmailSubjectParser):
+    """Custom Parser for Email subject of Tata circuit maintenance notifications."""
+
+    def parse_subject(self, subject: str) -> List[Dict]:
+        """Parse Tata Email subject for summary and status."""
+        data: Dict[str, Any] = {"summary": subject.strip().replace("\n", "")}
+        if "completion" in subject.lower():
```

Comment:
same idea to avoid computing lower() every time
```suggestion
        subject_lower = subject.lower()
        if "completion" in subject_lower:
```